### ISSUE TITLE
First attempt always fails - could not read from repo

### ISSUE THREAD

**Issue by yiga2** [kubernetes/git-sync#485]
I use `git-sync` 3.3.5 as a sidecar container on GKE and it always takes a (automatic) restart to make it work:
![image](https://user-images.githubusercontent.com/25224265/151826755-3fadd1f0-f4d9-4587-8025-60e3ee7ababf.png)

No big deal at all but I wonder why the first attempt to connect would fail. The logs do not tell exactly what failed (connection, credentials, repo existence,...) and github audit logs do not record access attempts.
Obviously I cannot look at more verbose local logs (in the container) as the failed container is terminated before I have a chance to look.

Any clue ? I don't mind adding a delay if that helps - but I don't see why it would.

**Comment by thockin** [comment]
That is odd.  You could try setting `--v=5` which will dump every command
it runs, maybe there's some difference between the first clone and the
second one?  Or maybe we could add a -v to `git clone`, though I am not
sure that would help.

**Comment by thockin** [comment]
any updates here?

**Comment by yiga2** [comment]
I have switched to 3.4 and not sure if the extent of what's captured by level 5 changed but now I see the actual reason:
![image](https://user-images.githubusercontent.com/25224265/152862672-c88fb5a2-e53f-4a5b-9569-f9640b1eaa9d.png)
Namely, `kubedns` has probably not fully started before `git-sync` tries to connect hence name resolution not possible.
Closing here as not a showstopper - it connects fine later  - and I will just add a slight delay before the `git-sync` container starts.
Confusing thought that git-sync (or git?) says _temporary_ name resolution error and _fatal_ error right after.
